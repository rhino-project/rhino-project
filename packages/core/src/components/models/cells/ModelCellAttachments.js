import { useCallback } from 'react';
import CellString from 'rhino/components/table/cells/CellString';
import { useGlobalOverrides } from 'rhino/hooks/overrides';

const defaultComponents = {
  ModelCellAttachments: CellString
};

const ModelCellAttachments = ({ overrides, getValue, ...props }) => {
  const { ModelCellAttachments } = useGlobalOverrides(
    defaultComponents,
    overrides
  );

  const syntheticGetValue = useCallback(
    () => (getValue()?.length ? `${getValue()?.length} files` : undefined),
    [getValue]
  );

  return (
    <ModelCellAttachments
      getValue={syntheticGetValue}
      {...props}
    ></ModelCellAttachments>
  );
};

export default ModelCellAttachments;
