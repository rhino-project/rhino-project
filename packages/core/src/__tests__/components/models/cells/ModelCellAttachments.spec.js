import { render } from '@testing-library/react';
import ModelCellAttachments from 'rhino/components/models/cells/ModelCellAttachments';
import { sharedCellTests } from './sharedCellTests';

describe('ModelCellAttachments', () => {
  sharedCellTests(ModelCellAttachments);
});
