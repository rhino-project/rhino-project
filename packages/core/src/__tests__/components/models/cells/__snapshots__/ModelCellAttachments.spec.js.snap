// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCellAttachments should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  -
</DocumentFragment>
`;

exports[`ModelCellAttachments should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  baz
</DocumentFragment>
`;

exports[`ModelCellAttachments should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
